src/NServiceBus.Metrics/ResetMetricTimer.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Threading.Tasks;
+using NServiceBus;
+using NServiceBus.Features;
+
+class ResetMetricTimer
+{
+    public event EventHandler NoMessageSentForAWhile;
+
+    public ResetMetricTimer(FeatureConfigurationContext featureConfigurationContext)
+    {
+        timer = new System.Threading.Timer(ResetCounterValueIfNoMessageHasBeenProcessedRecently, null, 0, 2000);
+
+        featureConfigurationContext.Pipeline.OnReceivePipelineCompleted(e =>
+        {
+            DateTime timeSent;
+            if (e.TryGetTimeSent(out timeSent))
+            {
+                lastMessageProcessedTime = e.CompletedAt;
+            }
+
+            return Task.FromResult(0);
+        });
+
+    }
+
+    void ResetCounterValueIfNoMessageHasBeenProcessedRecently(object state)
+    {
+        // Concurreny : This runs every 2s and might just fire before an update
+        if (NoMessageHasBeenProcessedRecently())
+        {
+            NoMessageSentForAWhile?.Invoke(this, null);
+        }
+    }
+
+    bool NoMessageHasBeenProcessedRecently()
+    {
+        var timeFromLastMessageProcessed = DateTime.UtcNow - lastMessageProcessedTime;
+        // Concurrency : This currently ignores messages being processed atm.
+        return timeFromLastMessageProcessed > estimatedMaximumProcessingDuration;
+    }
+
+    TimeSpan estimatedMaximumProcessingDuration = TimeSpan.FromSeconds(2);
+    DateTime lastMessageProcessedTime;
```

Comment:
Writing/reading from this fields happens on different threads and there are no memory barriers ensuring that the (reasonably) recent value of this will be used in estimating the condition. In this case making this fields `volatile` or using `Volatile.Read/Write` should be just fine. As far as I remember, you can't use volatile for complex types, hence you should convert it to ticks to make it.